src/msg/async/net_handler.cc

```diff
@@ -49,7 +49,7 @@ int NetHandler::create_socket(int domain, bool reuse_addr)
    * will be able to close/open sockets a zillion of times */
   if (reuse_addr) {
     int on = 1;
-    if (::setsockopt(s, SOL_SOCKET, SO_REUSEADDR, &on, sizeof(on)) == -1) {
+    if (::setsockopt(s, SOL_SOCKET, SO_REUSEADDR, (SOCKOPT_VAL_TYPE) &on, sizeof(on)) == -1) {
```

Comment:
You might want to look into difference in the networkingstack between windows and linux with regards of port/address reuse. These are very minute differences, but sometimes you cannot connect, because the statemachine in the stack has not finished, and fast reconnection becomes a problem. Resulting in long (secs) timeout.

Reply:
Things like:
[https://vincent.bernat.ch/en/blog/2014-tcp-time-wait-state-linux](url)
explained for windows can help

Reply:
Thanks for the link, that seems quite comprehensive.

Properly handling all the subtle differences between the network stacks will probably take a while.